core/src/main/java/org/kiwix/kiwixmobile/core/main/AddNoteDialog.kt

```diff
@@ -0,0 +1,475 @@
+
+  @Inject
+  lateinit var sharedPreferenceUtil: SharedPreferenceUtil
+
+  @Inject
+  lateinit var zimReaderContainer: ZimReaderContainer
+
+  @JvmField @Inject
+  var alertDialogShower: AlertDialogShower? = null
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    coreComponent
+      .activityComponentBuilder()
+      .activity(requireActivity())
+      .build()
+      .inject(this)
+
+    // Returns name of the form ".../Kiwix/granbluefantasy_en_all_all_nopic_2018-10.zim"
+    zimFileName = zimReaderContainer.zimCanonicalPath
+    if (zimFileName != null) { // No zim file currently opened
+      zimFileTitle = zimReaderContainer.zimFileTitle
+      articleTitle = (activity as WebViewProvider?)?.getCurrentWebView()?.title
+
+      // Corresponds to "ZimFileName" of "{External Storage}/Kiwix/Notes/ZimFileName/ArticleUrl.txt"
+      val zimNoteDirectoryName = zimNoteDirectoryName
+      articleNoteFileName = getArticleNotefileName()
+      zimNotesDirectory = "$NOTES_DIRECTORY$zimNoteDirectoryName/"
+    } else {
+      onFailureToCreateAddNoteDialog()
+    }
+  }
+
+  private fun onFailureToCreateAddNoteDialog() {
+    context.toast(R.string.error_file_not_found, Toast.LENGTH_LONG)
+    closeKeyboard()
+    requireFragmentManager().beginTransaction().remove(this).commit()
+  }
+
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View {
+    super.onCreateView(inflater, container, savedInstanceState)
+    root = inflater.inflate(R.layout.dialog_add_note, container, false)
+    return root
+  }
+
+  private val zimNoteDirectoryName: String
+    get() {
+      val noteDirectoryName = getTextAfterLastSlashWithoutExtension(zimFileName!!)
+      return (if (noteDirectoryName.isNotEmpty()) noteDirectoryName else zimFileTitle)!!
+    }
+
+  private fun getArticleNotefileName(): String {
+    // Returns url of the form: "content://org.kiwix.kiwixmobile.zim.base/A/Main_Page.html"
+    val articleUrl = (activity as WebViewProvider?)?.getCurrentWebView()?.url
+    var noteFileName = ""
+    if (articleUrl == null) {
+      onFailureToCreateAddNoteDialog()
+    } else {
+      noteFileName = getTextAfterLastSlashWithoutExtension(articleUrl)
+    }
+    return (if (noteFileName.isNotEmpty()) noteFileName else articleTitle)!!
+  }
+
+  private fun getTextAfterLastSlashWithoutExtension(path: String): String {
+    /* That's about exactly what it does.
+     *
+     * From ".../Kiwix/granbluefantasy_en_all_all_nopic_2018-10.zim", returns "granbluefantasy_en_all_all_nopic_2018-10"
+     * From "content://org.kiwix.kiwixmobile.zim.base/A/Main_Page.html", returns "Main_Page"
+     * For null input or on being unable to find required text, returns null
+     * */
+    val rightmostSlash = path.lastIndexOf('/')
+    val rightmostDot = path.lastIndexOf('.')
+    return if (rightmostSlash > -1 && rightmostDot > -1) {
```

Comment:
this can be rewritten with kotlin `substring` methods and will be much more readable

Reply:
How?

Reply:
```
 /* That's about exactly what it does.
     *
     * From ".../Kiwix/granbluefantasy_en_all_all_nopic_2018-10.zim", returns "granbluefantasy_en_all_all_nopic_2018-10"
     * From "content://org.kiwix.kiwixmobile.zim.base/A/Main_Page.html", returns "Main_Page"
     * For null input or on being unable to find required text, returns null
     * */
```

this scratch should put you on the right path
![image](https://user-images.githubusercontent.com/3358282/91834298-c1fe8880-ec3f-11ea-9407-47925009c9cb.png)

Reply:
```
path.substringAfterLast('/').substringBeforeLast('.', "")
```
@macgills is this the correct way?

Reply:
yup, perfectamundo

Reply:
hmmm actually maybe it should be `substringAfterLast('/', "")` also, just in case